share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorService.java

```diff
@@ -521,17 +525,30 @@ public void onElection(int partitionIndex, int partitionLeaderEpoch) {
 
     @Override
     public void onResignation(int partitionIndex, OptionalInt partitionLeaderEpoch) {
+        throwIfNotActive();
         runtime.scheduleUnloadOperation(
             new TopicPartition(Topic.SHARE_GROUP_STATE_TOPIC_NAME, partitionIndex),
             partitionLeaderEpoch
         );
     }
 
+    @Override
+    public void onNewMetadataImage(MetadataImage newImage, MetadataDelta delta) {
+        throwIfNotActive();
+        this.runtime.onNewMetadataImage(newImage, delta);
+    }
+
     private TopicPartition topicPartitionFor(SharePartitionKey key) {
         return new TopicPartition(Topic.SHARE_GROUP_STATE_TOPIC_NAME, partitionFor(key.toString()));
     }
 
     private static <P> boolean isEmpty(List<P> list) {
         return list == null || list.isEmpty();
     }
+
+    private void throwIfNotActive() {
```

Comment:
This method probably ought to be guarding the `readState` and `writeState` methods too.

Reply:
It is - line 284, 404
just that it is not being used directly but the check is the same. We are just wrapping the response as a completable future to easier handling at KafkaApis level